internal/bot/slashcommand/handler/report.go

```diff
@@ -18,12 +19,19 @@ import (
 	"github.com/bwmarrin/discordgo"
 )
 
-func handleNewReport(ctx context.Context, session *discordgo.Session, interaction *discordgo.InteractionCreate, client snitchv1connect.ReportServiceClient, userClient snitchv1connect.UserHistoryServiceClient) {
+func handleNewReport(ctx context.Context, session *discordgo.Session, interaction *discordgo.InteractionCreate, client snitchv1connect.ReportServiceClient, userClient snitchv1connect.UserHistoryServiceClient, tokenGenerator *auth.TokenGenerator) {
 	slogger, ok := ctxutil.Value[*slog.Logger](ctx)
 	if !ok {
 		slogger = slog.Default()
 	}
 
+	token, err := tokenGenerator.Generate(interaction.GuildID)
+	if err != nil {
+		slogger.ErrorContext(ctx, "failed to generate token", "error", err)
+		messageutil.SimpleRespondContext(ctx, session, interaction, "Failed to generate token.")
+		return
+	}
```

Comment:
Yeah, there's so much duplication here, would really like to see this stuff extracted and put into a middleware